### DISCUSSION TITLE
How to use thread pool in discord.py bot command?

### DISCUSSION THREAD

**Discussion by nttstar** [Rapptz/discord.py#9408] [Q&A] [answered]
Hello，I have a bot command that includes a method which is time-consuming and needs to be executed synchronously. I want to use a thread pool to save this execution object and execute it one by one in another executor. What is the best implementation method?

**Comment by Rapptz** [comment, marked as answer]
`asyncio` already comes with an executor concept, [`Loop.run_in_executor`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor) so that should be used.

**Comment by nttstar** [comment]
Thanks @Rapptz , the ThreadPoolExecutor class works!
